src/NServiceBus.Testing/TestableMessageHandlerContext.cs

```diff
@@ -0,0 +1,162 @@
+namespace NServiceBus.Testing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Threading.Tasks;
+    using NServiceBus.Extensibility;
+    using NServiceBus.Persistence;
+    using NServiceBus.Testing.ExpectedInvocations;
+
+    internal class TestableMessageHandlerContext : IMessageHandlerContext
```

Comment:
I think we'll make this one public as a separate PR since this class seems to need some more work in order to achieve that.